services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/dataaccess/AccessRequirementManagerImplUnitTest.java

```diff
@@ -956,4 +1084,81 @@ public void testValidateAccessRequirement() {
 		AccessRequirement ar = createExpectedAR();
 		AccessRequirementManagerImpl.validateAccessRequirement(ar);
 	}
+
+	@Test
+	public void testSignalDeletedSubjectIdsAll() {
+		List<RestrictableObjectDescriptor> currentRods = generateRods(2);
+		List<RestrictableObjectDescriptor> updatedRods = new ArrayList<>();
+		// call under test: should all be signaled
+		arm.signalDeletedSubjectIds(currentRods, updatedRods);
+		verify(nodeDao, times(2)).getNodeTypeById(any(String.class));
+		verify(nodeDao).getNodeTypeById("0");
+		verify(nodeDao).getNodeTypeById("1");
+		verify(mockTransactionalMessenger, times(2)).sendMessageAfterCommit(any(String.class), eq(ObjectType.ENTITY), eq(ChangeType.UPDATE));
+		verify(mockTransactionalMessenger).sendMessageAfterCommit(eq("0"), eq(ObjectType.ENTITY), eq(ChangeType.UPDATE));
+		verify(mockTransactionalMessenger).sendMessageAfterCommit(eq("1"), eq(ObjectType.ENTITY), eq(ChangeType.UPDATE));
+	}
+
+	@Test
+	public void testSignalDeletedSubjectIdsNone() {
+		List<RestrictableObjectDescriptor> updatedRods = generateRods(2);
+		List<RestrictableObjectDescriptor> currentRods = new ArrayList<>();
+		// call under test: none should be signaled
+		arm.signalDeletedSubjectIds(currentRods, updatedRods);
+		verify(nodeDao, never()).getNodeTypeById(any(String.class));
+		verify(mockTransactionalMessenger, never()).sendMessageAfterCommit(any(String.class), any(ObjectType.class), any(ChangeType.class));
+	}
+
+	@Test
+	public void testSignalDeletedSubjectIdsOne() {
+		List<RestrictableObjectDescriptor> currentRods = generateRods(2); 	// 0,1
+		List<RestrictableObjectDescriptor> updatedRods = generateRods(1);	//	0
+		when(nodeDao.getNodeTypeById("1")).thenReturn(EntityType.file);
+		// call under test: "1" should be signaled
+		arm.signalDeletedSubjectIds(currentRods, updatedRods);
+		verify(nodeDao).getNodeTypeById("1");
+		verify(mockTransactionalMessenger).sendMessageAfterCommit("1", ObjectType.ENTITY, ChangeType.UPDATE);
+	}
+
+	@Test
+	public void testSignalDeletedSubjectIdsOneDeleted() {
+		List<RestrictableObjectDescriptor> currentRods = generateRods(1); // id == 0
+		List<RestrictableObjectDescriptor> updatedRods = new ArrayList<>();
+		RestrictableObjectDescriptor rod = new RestrictableObjectDescriptor();
+		rod.setId("1");
+		rod.setType(RestrictableObjectType.ENTITY);
+		updatedRods.add(rod);
+		when(nodeDao.getNodeTypeById("0")).thenThrow(new NotFoundException());
+		// call under test: "0" should all be signaled
```

Comment:
Comment is incorrect. In this case we catch the NotFound and return b4 sending msg.